Scripts/dea_datahandling.py

```diff
@@ -704,3 +714,202 @@ def paths_to_datetimeindex(paths, string_slice=(0, 10)):
     date_strings = [os.path.basename(i)[slice(*string_slice)]
                     for i in paths]
     return pd.to_datetime(date_strings)
+
+
+def calc_geomedian(ds,
+                   axis="time",
+                   max_value=None,
+                   min_value=None,
+                   num_threads=1,
+                   eps=1e-7,
+                   nocheck=True):
+    '''
+    Runs geomedian over a xarray.Dataset or xarray.DataArray.
+
+    Specify min_value and max_value if known for better performance,
+    especially if using dask.
+
+    Parameters
+    ----------
+    da : xarray.DataArray object
+    max_value : the maximum value that could be found in the array
+    min_value : the minimum value that could be found in the array
+    '''
+    da = (odc.algo.reshape_for_geomedian(ds, axis=axis) if
+          isinstance(ds, xr.Dataset) else ds)
```

Comment:
`xr_geomedian` does accept `xr.Dataset` objects now and performs reshape internally.

Reply:
Thanks @Kirill888, can you update the code in this PR to reflect this change? And then I will port it across to DE Africa.

Reply:
I think `odc.algo.int_geomedian` is exactly this function, but has different internal implementation that takes advantage of knowing that input is `int`.

`int_geomedian` does assume `time,y,x` dimension order, but all of our data is like that, and instead of min/max it takes scale offset with default being no scale no offset

Reply:
hold on, in the example notebook loaded data is already `float32` cause this happened inside `load_ard`

Reply:
Ah yes, if the data was loaded in `load_ard` with cloud masking selected it will be in `float32`

Reply:
I think we should be able to get it back as `Int` if we add `dtype='native'` to the function call

Reply:
I reckon staying in `int16` throughout is preferable.
`xr_geomedian` for float inputs, `int_geomedian` for int16 inputs, ideally I should merge it into one function though.

Reply:
Having them as one function would be really nice, `xr_geomedian` is a nice name for a general purpose wrapper function I think